src/kOS/Module/kOSVesselModule.cs

```diff
@@ -234,7 +235,48 @@ private void AddDefaultParameters()
             AddFlightControlParameter("wheelthrottle", new WheelThrottleManager(Vessel));
             AddFlightControlParameter("flightcontrol", new FlightControl(Vessel));
             flightParametersAdded = true;
-            SafeHouse.Logger.SuperVerbose(string.Format("kOSVesselModule AddDefaultParameters()!  On {0}", Vessel.vesselName));
+            SafeHouse.Logger.SuperVerbose(string.Format("kOSVesselModule AddDefaultParameters()!  On {0}({1}", Vessel.vesselName, Vessel.id));
+        }
+
+        /// <summary>
+        /// After a decouple or part explosion, it's possible for this vessel to
+        /// still have an assigned flight control parameter that is coming from
+        /// a kOS core that is no longer on this vessel but is instead on the newly
+        /// branched vessel we left behind.  If so, that parameter needs to be
+        /// removed from this vessel.  The new kOSVesselModule will take care of
+        /// making a new parameter on the new vessel, but this kOSVesselModule needs
+        /// to detach it from this one.
+        /// </summary>
+        private void ResetPhysicallyDetachedParameters()
+        {
+            List<string> removeKeys = new List<string>();
+            foreach (string key in flightControlParameters.Keys)
+            {
+                IFlightControlParameter p = flightControlParameters[key];
+                if (p.GetShared() != null && p.GetShared().Vessel != null && vessel != null &&
+                    p.GetShared().Vessel.id != vessel.id)
+                    removeKeys.Add(key);
+            }
+            foreach (string key in removeKeys)
+            {
+                SafeHouse.Logger.SuperVerbose(string.Format(
+                    "kOSVesselModule: re-defaulting parameter \"{0}\" because it's on a detached part of the vessel.", key));
+                RemoveFlightControlParameter(key);
+                IFlightControlParameter p = null;
+                if (key.Equals("steering"))
+                    p = new SteeringManager(Vessel);
+                else if (key.Equals("throttle"))
+                    p = new ThrottleManager(Vessel);
+                else if (key.Equals("wheelsteering"))
+                    p = new WheelSteeringManager(Vessel);
+                else if (key.Equals("wheelthrottle"))
+                    p = new WheelThrottleManager(Vessel);
+                else if (key.Equals("flightcontrol"))
+                    p = new FlightControl(Vessel);
+
+                if (p != null)
+                    AddFlightControlParameter(key, p);
+            }
```

Comment:
The reason I have two loops - the first to find the keys to remove and the second to remove them - instead of just one loop is that I couldn't remember if .Net's rules that ban deleting from a collection while iterating over it counted for iterating the keys of a Dictionary.  It doesn't make any sense for them to since the keys are lookups rather than integer index that would change position, but a lot of what .Net bans when it comes to this don't make sense to me anyway so it seemed like it might ban it.